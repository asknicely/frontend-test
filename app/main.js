import Vue from "vue";
import App from "./App.vue";
import store from "./store";
import styles from "./styles/index.css";
import 'animate.css';

/* import the fontawesome core */
import { library } from "@fortawesome/fontawesome-svg-core";

/* import specific icons */
import { faTrashCan, faCheck, faPlus } from "@fortawesome/free-solid-svg-icons";

/* import font awesome icon component */
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";

/* add icons to the library */
library.add(faTrashCan);
library.add(faCheck);
library.add(faPlus);

/* add font awesome icon component */
Vue.component("font-awesome-icon", FontAwesomeIcon);

Vue.config.productionTip = false;

window.onload = function () {
  new Vue({
    styles,
    store,
    render: (h) => h(App),
  }).$mount("#vue-app");
};
