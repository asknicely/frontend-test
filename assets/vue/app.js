/*/////  Light Vue Todos for Symfony2.0 /////
///////  Author: Kinson Lau ///////////////////////// */
import Vue from "vue";
import Todos from "./views/Todos";
import Todo from "./views/Todo";

/*
  VUE CONFIG
*/
Vue.config.devtools = true; // Remove for production
Vue.config.productionTip = false; // Update for production

/*
  ROOT COMPONENT
  METHODS
  - addToCart: Adds product via the product page and product tiles
  - updateQuantity: Updates an item in the cart
  - setFocus: Displays / hides the mobile search form and focuses
*/

new Vue({
  delimiters: ["${", "}"],
  el: "#app",
  components: {
    Todos: Todos,
    Todo: Todo
  }
});
