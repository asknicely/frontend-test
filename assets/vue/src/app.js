/* /////  Light Vue Todos - Symfony2.0 /////
///////  Author: Kinson Lau ///////////////////////// */

/*
  Import Vue components
*/
import Vue from 'vue'
import Todos from './views/Todos'

/*
  VUE CONFIG
*/
Vue.config.devtools = true // Remove for production
Vue.config.productionTip = false // Update for production

/*
  Init vue app
*/
new Vue({ // eslint-disable-line no-new
  delimiters: ['${', '}'],
  el: '#app',
  components: {
    Todos: Todos
  }
})
