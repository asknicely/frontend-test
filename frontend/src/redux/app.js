import React from 'react'

export const Context = React.createContext()

export const initial = {
    username: '',
    isLoggedIn: false,
    token: ''
}

export const reducer = (state, action) => {
    switch (action.type) {

        case 'DO_LOGIN':
            return { 
                ...state, 
                username: action.username,
                token: action.token,
                isLoggedIn: true
            }

        case 'DO_LOGOUT':
            return { 
                ...state, 
                username: '',
                token: '',
                isLoggedIn: false
            }            
        
        default:
            return {
                ...state,
            }
    }
}