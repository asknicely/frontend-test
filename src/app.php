<?php

use Silex\Application;
use Silex\Provider\SessionServiceProvider;
use Silex\Provider\TwigServiceProvider;
use Silex\Provider\UrlGeneratorServiceProvider;
use Silex\Provider\ValidatorServiceProvider;
use Silex\Provider\ServiceControllerServiceProvider;
use Silex\Provider\HttpFragmentServiceProvider;
use Silex\Provider\DoctrineServiceProvider;
use DerAlex\Silex\YamlConfigServiceProvider;
use Symfony\Component\HttpFoundation\Request;
use Symfony\Component\HttpFoundation\Response;

$app = new Application();
$app->register(new SessionServiceProvider());
$app->register(new UrlGeneratorServiceProvider());
$app->register(new ValidatorServiceProvider());
$app->register(new ServiceControllerServiceProvider());
$app->register(new TwigServiceProvider());
$app->register(new HttpFragmentServiceProvider());

$app->register(new YamlConfigServiceProvider(__DIR__.'/../config/config.yml'));
$app->register(new DoctrineServiceProvider, array(
    'db.options' => array(
        'driver'    => 'pdo_mysql',
        'host'      => $app['config']['database']['host'],
        'dbname'    => $app['config']['database']['dbname'],
        'user'      => $app['config']['database']['user'],
        'password'  => $app['config']['database']['password'],
        'charset'   => 'utf8',
    ),
));

$app->before(function (Request $request) use ($app) {
    if ($request->isMethod('OPTIONS')) {
        return new Response();
    }
});

$app->after(function (Request $request, Response $response) {
    $response->headers->set('Access-Control-Allow-Origin', $request->headers->get('origin'));
    $response->headers->set('Access-Control-Allow-Methods', 'GET, POST, PUT, PATCH, DELETE, OPTIONS');
    $response->headers->set('Access-Control-Allow-Credentials', 'true');
});

return $app;
