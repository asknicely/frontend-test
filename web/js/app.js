/*/////  Light Vue Todos for Symfony2.0 /////
///////  Author: Kinson Lau ///////////////////////// */

/*
  VUE CONFIG
*/
Vue.config.devtools = true // Remove for production
Vue.config.productionTip = false // Update for production

/* 
  EVENTS CENTRE
  This vue instance is used to dispatch and recieve events
  - Use $bus.$emit to send an event
  - Use $bus.$on to recieve an event
*/
var bus = new Vue({})
Object.defineProperty(Vue.prototype, '$bus', {
  get() {
    return this.$root.bus
  },
  set(newInstance) {
    this.$root.bus = newInstance
  }
})


/*
  ROOT COMPONENT
  METHODS
  - addToCart: Adds product via the product page and product tiles
  - updateQuantity: Updates an item in the cart
  - setFocus: Displays / hides the mobile search form and focuses
*/

import Todos from './vue/components/Todos.js'

new Vue({
  delimiters: ['${', '}'],
  el: '#app',
  data: {
    bus: new Vue()
  },
  components: {
    VueTodos: Todos
  }
})