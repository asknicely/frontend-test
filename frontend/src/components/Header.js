import React, { useContext } from 'react'
import { Link } from "react-router-dom"
import { Context } from '../redux/app'

const Header = (props) => {

	const { state } = useContext(Context)

	return (
		<nav className="navbar navbar-inverse navbar-fixed-top">
			<div className="container">
				<div className="navbar-header">
					<Link to="/" className="navbar-brand">
						<span className="glyphicon glyphicon-home glyphicon-white"></span>
					</Link>
				</div>
				<div id="navbar" className="collapse navbar-collapse">
					<ul className="nav navbar-nav">
						<li><Link to="/todo">Todo list</Link></li>
					</ul>
					<ul className="nav navbar-nav navbar-right">
						
						<li style={{ display: state.token ? 'block' : 'none '}}>
							<Link to="/logout">

								<span className="glyphicon glyphicon-user custom-span"></span>
								{state.username} 
								
								<span className="glyphicon glyphicon-off custom-span"></span> 
								Logout

							</Link>
						</li>
					
						<li style={{ display: state.token ? 'none' : 'block'}}>
							<Link to="/login">Login</Link>
						</li>
						
					</ul>
				</div>
			</div>
		</nav>
	)
}

export default Header