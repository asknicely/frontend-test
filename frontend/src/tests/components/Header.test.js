import React, { useReducer } from 'react';
import { Router } from "react-router-dom";
import { shallow, mount, render } from 'enzyme';
import Header from '../../components/Header';
import { initial, Context, reducer } from '../../redux/app'
import { history } from '../functions/History'


test('should render Header correctly and display LOGIN link ', () => {

	const state = {}
	const App = (props) => {
		// const [state, dispatch] = useReducer(reducer, initial)
		return (
			<Context.Provider value={{ state }}>
				<Router history={history}>
					<Header />
				</Router>
			</Context.Provider>
		);
	}

	const wrapper = mount(<App />);
	expect(wrapper).toMatchSnapshot();
	
});


test('should render Header correctly and display STATE.username and LOGOUT link ', () => {

	const state = { username: 'User123' }
	const App = (props) => {
		// const [state, dispatch] = useReducer(reducer, initial)
		return (
			<Context.Provider value={{ state }}>
				<Router history={history}>
					<Header />
				</Router>
			</Context.Provider>
		);
	}

	const wrapper = mount(<App />);
	expect(wrapper).toMatchSnapshot();
	
});
