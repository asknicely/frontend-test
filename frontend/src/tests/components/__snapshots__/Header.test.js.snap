// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly and display LOGIN link  1`] = `
<App>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Header>
      <nav
        className="navbar navbar-inverse navbar-fixed-top"
      >
        <div
          className="container"
        >
          <div
            className="navbar-header"
          >
            <Link
              className="navbar-brand"
              to="/"
            >
              <LinkAnchor
                className="navbar-brand"
                href="/"
                navigate={[Function]}
              >
                <a
                  className="navbar-brand"
                  href="/"
                  onClick={[Function]}
                >
                  <span
                    className="glyphicon glyphicon-home glyphicon-white"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </div>
          <div
            className="collapse navbar-collapse"
            id="navbar"
          >
            <ul
              className="nav navbar-nav"
            >
              <li>
                <Link
                  to="/todo"
                >
                  <LinkAnchor
                    href="/todo"
                    navigate={[Function]}
                  >
                    <a
                      href="/todo"
                      onClick={[Function]}
                    >
                      Todo list
                    </a>
                  </LinkAnchor>
                </Link>
              </li>
            </ul>
            <ul
              className="nav navbar-nav navbar-right"
            >
              <li
                style={
                  Object {
                    "display": "none ",
                  }
                }
              >
                <Link
                  to="/logout"
                >
                  <LinkAnchor
                    href="/logout"
                    navigate={[Function]}
                  >
                    <a
                      href="/logout"
                      onClick={[Function]}
                    >
                      <span
                        className="glyphicon glyphicon-user custom-span"
                      />
                      <span
                        className="glyphicon glyphicon-off custom-span"
                      />
                      Logout
                    </a>
                  </LinkAnchor>
                </Link>
              </li>
              <li
                style={
                  Object {
                    "display": "block",
                  }
                }
              >
                <Link
                  to="/login"
                >
                  <LinkAnchor
                    href="/login"
                    navigate={[Function]}
                  >
                    <a
                      href="/login"
                      onClick={[Function]}
                    >
                      Login
                    </a>
                  </LinkAnchor>
                </Link>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </Header>
  </Router>
</App>
`;

exports[`should render Header correctly and display STATE.username and LOGOUT link  1`] = `
<App>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Header>
      <nav
        className="navbar navbar-inverse navbar-fixed-top"
      >
        <div
          className="container"
        >
          <div
            className="navbar-header"
          >
            <Link
              className="navbar-brand"
              to="/"
            >
              <LinkAnchor
                className="navbar-brand"
                href="/"
                navigate={[Function]}
              >
                <a
                  className="navbar-brand"
                  href="/"
                  onClick={[Function]}
                >
                  <span
                    className="glyphicon glyphicon-home glyphicon-white"
                  />
                </a>
              </LinkAnchor>
            </Link>
          </div>
          <div
            className="collapse navbar-collapse"
            id="navbar"
          >
            <ul
              className="nav navbar-nav"
            >
              <li>
                <Link
                  to="/todo"
                >
                  <LinkAnchor
                    href="/todo"
                    navigate={[Function]}
                  >
                    <a
                      href="/todo"
                      onClick={[Function]}
                    >
                      Todo list
                    </a>
                  </LinkAnchor>
                </Link>
              </li>
            </ul>
            <ul
              className="nav navbar-nav navbar-right"
            >
              <li
                style={
                  Object {
                    "display": "none ",
                  }
                }
              >
                <Link
                  to="/logout"
                >
                  <LinkAnchor
                    href="/logout"
                    navigate={[Function]}
                  >
                    <a
                      href="/logout"
                      onClick={[Function]}
                    >
                      <span
                        className="glyphicon glyphicon-user custom-span"
                      />
                      User123
                      <span
                        className="glyphicon glyphicon-off custom-span"
                      />
                      Logout
                    </a>
                  </LinkAnchor>
                </Link>
              </li>
              <li
                style={
                  Object {
                    "display": "block",
                  }
                }
              >
                <Link
                  to="/login"
                >
                  <LinkAnchor
                    href="/login"
                    navigate={[Function]}
                  >
                    <a
                      href="/login"
                      onClick={[Function]}
                    >
                      Login
                    </a>
                  </LinkAnchor>
                </Link>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </Header>
  </Router>
</App>
`;
